File: apps/api/src/routes/auditLogs.ts
import { Router, Response } from 'express';
import { AuthRequest, authenticateRequest } from '../middleware/auth';
import { listAuditLogs } from '../database';

const router = Router();
router.use(authenticateRequest);

router.get('/', async (req: AuthRequest, res: Response) => {
  if (!req.tenantId) return res.status(401).json({ error: 'Unauthorized' });

  const limit = parseInt(String(req.query.limit || '50'), 10);
  const logs = await listAuditLogs(req.tenantId, Number.isFinite(limit) ? limit : 50);
  return res.json({ auditLogs: logs });
});

export default router;

File: apps/api_python/routes/audit_logs.py
from fastapi import APIRouter, Depends, Query

import database as db
from auth import AuthContext, authenticate_request

router = APIRouter()

from datetime import datetime

@router.get('')
async def list_logs(
    auth: AuthContext = Depends(authenticate_request),
    limit: int = Query(default=50, ge=1, le=200),
    resourceType: str | None = Query(default=None),
    action: str | None = Query(default=None),
    userId: str | None = Query(default=None),
    fromDate: datetime | None = Query(default=None),
    toDate: datetime | None = Query(default=None),
    cursor: str | None = Query(default=None),
):
    logs = db.list_audit_logs(
        auth.tenant_id,
        limit=limit,
        resource_type=resourceType,
        action=action,
        user_id=userId,
        from_date=fromDate,
        to_date=toDate,
        cursor=cursor
    )
    total = db.count_audit_logs(
        auth.tenant_id,
        resource_type=resourceType,
        action=action,
        user_id=userId,
        from_date=fromDate,
        to_date=toDate
    )
    next_cursor = logs[-1]['id'] if len(logs) == limit else None
    return {'auditLogs': logs, 'nextCursor': next_cursor, 'total': total}
